File: leetcode/1266.py
class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        res = 0
        for i in range(len(points)-1):
            A = points[i]
            B = points[i+1]
            res += self.shortestFromAtoB(A, B)
        return res

    def shortestFromAtoB(self, A, B):
        x1, y1 = A
        x2, y2 = B
        return max(abs(x1-x2), abs(y1-y2))

File: leetcode/1266.rb
# @param {Integer[][]} points
# @return {Integer}
def min_time_to_visit_all_points(points)
    res = 0
    (0...points.length-1).each do |i|
        x1, y1 = points[i]
        x2, y2 = points[i+1]
        res += [(x1-x2).abs, (y1-y2).abs].max
    end
    res
end